packages/core/src/components/axes/axis.ts

```diff
@@ -414,6 +432,51 @@ export class Axis extends Component {
 			container.attr("opacity", 0);
 		}
 
+		// truncate the label if it's too long
+		// only applies to discrete type
+		if (truncationType !== TruncationTypes.NONE && !isTimeScaleType && axisOptions.scaleType === "labels") {
```

Comment:
Can these checks be stored in a variable? They're fairly long, and we should use Tools.getProperty() to avoid the double checks (axisOptions.scaleType && axisOptions.scaleType === "labels")

also we might not need `!isTimeScaleType` because we're already checking if `axisOptions.scaleType === "labels"`

Reply:
also `"labels"` should say `ScaleTypes.LABELS`

Reply:
Resolved. Thanks.